plutus-ir/test/errors/mutuallyRecursiveTypes.golden

```diff
@@ -1 +1,3 @@
-Unsupported construct: Mutually recursive datatypes ((recursive) let binding; from mutuallyRecursiveTypes:1:2)
\ No newline at end of file
+Unsupported construct: Mutually recursive datatypes ((recursive) let binding; from [ <unknown>
+, mutuallyRecursiveTypes:1:2
+, mutuallyRecursiveTypes:1:2 ])
```

Comment:
I removed the unknown, but I think the duplicates are happening because of the way we replicate annotations among members of a let group (see `Letfloat.RhsTable`). We separate all members and add them in the `RhsTable`, then when we try to place the lets back (float them), the same annotation will appear multiple times. Should we consider `MultipleSources (Set Provenance a)` instead?